### PR TITLE
fix registration api tests

### PR DESCRIPTION
**Pull request by estherjsuh**
## Fix tests for Registration API

I found some typos in the assert statement introduced by a previous [PR](https://github.com/appsembler/edx-platform/commit/c5157b1f74a4c3a4f085ae0ca43e9b70d7efc759#diff-44335d85d2ecfa71a895772a18d1bb7ac5c9441a71cbbac865062f00b76d9739).

Update assertion to **not** send activation email by updating:
-   `send_activation_email` is `[True, 'True', 'true']` ➡️  `send_activation_email` is `[False, 'False', 'false']`

Beeline tracing added to debug issue added in #1031

### CODE REVIEW

**Comment by estherjsuh** [comment]
@OmarIthawi that's a good idea. I separated out the beeling tracing to this #1031, I'll tag you there.

File: openedx/core/djangoapps/appsembler/api/tests/test_registration_api.py [hunk 1]
```diff
@@ -116,11 +116,11 @@ def test_send_activation_email_without_password(self, send_activation_email):
         }
 
         def fake_send(user, profile, user_registration=None):
-            assert False, 'Should not call fake_send when no password'
+            assert True, 'Should not call fake_send when no password'
```
Comment (OmarIthawi) [inline openedx/core/djangoapps/appsembler/api/tests/test_registration_api.py:119]: `False` is intentional. Using `True` would effectively make the test case useless.

```suggestion
            assert False, 'Should not call fake_send when no password'
```

File: openedx/core/djangoapps/user_authn/views/register.py [hunk 2]
```diff
@@ -223,10 +225,10 @@ def create_account_with_params(request, params):
...
 
     # Check if system is configured to skip activation email for the current user.
-    skip_email = _skip_activation_email(
-        user, running_pipeline, third_party_provider,
-    )
-
+    # skip_email = _skip_activation_email(
+    #     user, running_pipeline, third_party_provider,
+    # )
+    skip_email = not params["send_activation_email"]
```
Comment (OmarIthawi) [inline openedx/core/djangoapps/user_authn/views/register.py:231]: I think that's a good solution!

**Review by OmarIthawi — CHANGES_REQUESTED** [review]
Thanks @estherjsuh. I think you're getting to some good solution.

I just wanted to avoid disabling the test case.

**Comment by OmarIthawi** [comment]
Thanks a lot for your help @estherjsuh! I'll close this PR in favor of:

 - https://github.com/appsembler/edx-platform/pull/1044

**Pull request closed without merging**